xray/client/xray_viewer.ts

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+function matrixToString(m: THREE.Matrix4): string {
+  const me = m.elements;
+  return [
+    me[0].toFixed(10),
+    me[1].toFixed(10),
+    me[2].toFixed(10),
+    me[3].toFixed(10),
+    me[4].toFixed(10),
+    me[5].toFixed(10),
+    me[6].toFixed(10),
+    me[7].toFixed(10),
+    me[8].toFixed(10),
+    me[9].toFixed(10),
+    me[10].toFixed(10),
+    me[11].toFixed(10),
+    me[12].toFixed(10),
+    me[13].toFixed(10),
+    me[14].toFixed(10),
+    me[15].toFixed(10),
+  ].join(',');
```

Comment:
Why not `me.map(x => x.toFixed(10)).join(',')`?

Reply:
This does not work, since the type is a Float32Array and map there is expected to return more values for another Float32Array.

Typescript was actually correct in warning us :)

Reply:
I see, a JS quirk really, that a Float32Array is not an Array. Wow, even `me.reduce((arr, x) => arr.push(x), [])` doesn't work. That's kinda funny.

Well, then what about:

```
const arr = [];
m.elements.forEach(x => arr.push(x.toFixed(10)));
return arr.join(',');
```

Reply:
sorry. this slipped my attention - github code review is not great.